services/supervisor/src/transactions.rs

```diff
@@ -103,24 +99,57 @@ impl ValidateInput for DeployConfirmation {
     }
 }
 
-impl ValidateInput for StartService {
-    type Error = ExecutionError;
-
-    fn validate(&self) -> Result<(), Self::Error> {
+impl StartService {
+    fn validate(&self, context: &CallContext<'_>) -> Result<(), ExecutionError> {
         self.artifact
             .validate()
             .map_err(|e| (Error::InvalidArtifactId, e))?;
         InstanceSpec::is_valid_name(&self.name)
             .map_err(|e| (Error::InvalidInstanceName, e))
-            .map_err(Self::Error::from)
+            .map_err(ExecutionError::from)?;
+
+        // Check that artifact is deployed.
+        if context
```

Comment:
Nit: it could make sense to extract `context.data().for_dispatcher()` into a variable in order to improve readability.